### PR TITLE
[nix] Add flake support to wander

### PR DESCRIPTION
**Pull request by dbalan**
This commit lets flake users import wander package directly from the repo, as well as quickly run wander with

```
nix run "github:robinovitch61/wander"
```

### CODE REVIEW

**Comment by robinovitch61** [comment]
This is awesome! Thank you @dbalan :) Question: the version shown on wander is `devel` when run with nix flakes, which means "unavailable":

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    flake-utils.lib.eachDefaultSystem (system:
+      let pkgs = nixpkgs.legacyPackages.${system};
+      in {
+        packages.default = self.packages.${system}.wander;
+
+        packages.wander = pkgs.buildGoModule {
+          name = "wander";
+          vendorHash = "sha256-SqDGXV8MpvEQFAkcE1NWvWjdzYsvbO5vA6k+hpY0js0=";
+          src = ./.;
+
```
Comment (dbalan) [inline flake.nix:19]:
```suggestion
          src = ./.;
         
          ldflags =
            [ "-X github.com/robinovitch61/wander/cmd.Version=v0.14.0" ];
```

The way to plug in version would be to add it to the build spec like above. I don't think there is any way to plug this in dynamically.

By default, `nix flake` just grabs whatever commit `main` is on, and pins that version in `flake.lock`. If we have a workflow to keep bumping this on releases, the actual code would but the release tag and commits on top.

How do you feel about it? The other option is to set the version for flake to something static like, `v{version}-flake` or just `flake`.

Comment (dbalan) [reply flake.nix:19]: ping @robinovitch61

Comment (robinovitch61) [reply flake.nix:19]: Ah sorry for the delay, for some reason wasn't getting emails about this! Gonna merge as is and see what I can do about the version embed later, thanks so much for this :)

**Review by robinovitch61 — APPROVED** [review]

**Pull request merged**